core/src/main/scala/com/eharmony/spotz/backend/ParallelFunctions.scala

```diff
@@ -0,0 +1,82 @@
+package com.eharmony.spotz.backend
+
+import com.eharmony.spotz.objective.Objective
+import com.eharmony.spotz.optimizer.grid.Grid
+import com.eharmony.spotz.optimizer.hyperparam.RandomSampler
+
+import scala.reflect.ClassTag
+
+/**
+  * Use scala parallel collections to parallelize the optimizer computation.
+  */
+trait ParallelFunctions extends BackendFunctions {
```

Comment:
Seems like in this class you could have a 3rd private function to reduce code duplication.  You pass in all the same parameters except have a function that returns a point (either by grid or sampling).